client.go

```diff
@@ -642,40 +664,101 @@ func makeRequestData(query string, options *QueryOptions) ([]byte, error) {
 	return r, nil
 }
 
-func columnsToImportRequest(indexName string, fieldName string, shard uint64, records []Record) *pbuf.ImportRequest {
-	rowIDs := make([]uint64, 0, len(records))
-	columnIDs := make([]uint64, 0, len(records))
-	timestamps := make([]int64, 0, len(records))
-	for _, record := range records {
-		column := record.(Column)
-		rowIDs = append(rowIDs, column.RowID)
-		columnIDs = append(columnIDs, column.ColumnID)
-		timestamps = append(timestamps, column.Timestamp)
+func columnsToImportRequest(field *Field, shard uint64, records []Record) *pbuf.ImportRequest {
+	var rowIDs []uint64
+	var columnIDs []uint64
+	var rowKeys []string
+	var columnKeys []string
+
+	recordCount := len(records)
+	timestamps := make([]int64, 0, recordCount)
+	hasRowKeys := field.options.keys
+	hasColKeys := field.index.options.keys
+
+	if hasRowKeys {
+		rowKeys = make([]string, 0, recordCount)
+	} else {
+		rowIDs = make([]uint64, 0, recordCount)
 	}
+
+	if hasColKeys {
+		columnKeys = make([]string, 0, recordCount)
+	} else {
+		columnIDs = make([]uint64, 0, recordCount)
+	}
+
+	if hasRowKeys && hasColKeys {
```

Comment:
I'm thinking this would be more readable as a single `for` loop with two if/else blocks inside (similar to the above two for making the slices).

Reply:
I agree, it would be more readable. But I've unrolled the for loop there to make it faster.

Reply:
The non-scientific benchmark I've ran with the following script shows that unrolled version is about 1 seconds faster for a batch of 200 million on my machine but simplified this and its field values counterpart anyway.